pyvista/core/filters/poly_data.py

```diff
@@ -2747,6 +2751,12 @@ def extrude_rotate(
                 PyvistaFutureWarning,
             )
 
+        if (
+            not isinstance(rotation_axis, (np.ndarray, collections.abc.Sequence))
+            or len(rotation_axis) != 3
+        ):
+            raise TypeError('Vector must be a length three vector')
```

Comment:
I'm not sure if this is `TypeError`, but I don't have a strong thought on this.
```suggestion
            raise ValueError('Vector must be a length three vector')
```

Reply:
Good. Let's follow it.